src/unix/process.c

```diff
@@ -348,25 +358,575 @@ static void uv__process_child_init(const uv_process_options_t* options,
 #endif
 
 
+#if defined(__APPLE__)
+typedef struct uv__posix_spawn_fncs_tag {
+  struct {
+    int (*addchdir_np)(const posix_spawn_file_actions_t *, const char *);
+  } file_actions;
+} uv__posix_spawn_fncs_t;
+
+
+static uv_once_t posix_spawn_init_once = UV_ONCE_INIT;
+static uv__posix_spawn_fncs_t posix_spawn_fncs;
+static int posix_spawn_can_use_setsid;
+
+
+static void uv__spawn_init_posix_spawn_fncs(void) {
+  /* Try to locate all non-portable functions at runtime */
+  posix_spawn_fncs.file_actions.addchdir_np =
+    dlsym(RTLD_DEFAULT, "posix_spawn_file_actions_addchdir_np");
+}
+
+
+static void uv__spawn_init_can_use_setsid(void) {
+  static const int MACOS_CATALINA_VERSION_MAJOR = 19;
+  char version_str[256];
+  char* version_major_str;
+  size_t version_str_size = 256;
+  int r;
+  int version_major;
+
+  /* Get a version string */
+  r = sysctlbyname("kern.osrelease", version_str, &version_str_size, NULL, 0);
+  if (r != 0)
+    return;
+
+  /* Try to get the major version number. If not found
+   * fall back to the fork/exec flow */
+  version_major_str = strtok(version_str, ".");
+  if (version_major_str == NULL)
+    return;
+
+  /* Parse the version major as a number. If it is greater than
+   * the major version for macOS Catalina (aka macOS 10.15), then
+   * the POSIX_SPAWN_SETSID flag is available */
+  version_major = atoi_l(version_major_str, LC_C_LOCALE);
```

Comment:
From CI:
```
/Users/iojs/build/workspace/libuv-test-commit-osx-cmake/nodes/osx1014/src/unix/process.c:403:45: error: use of undeclared identifier 'LC_C_LOCALE'
  version_major = atoi_l(version_major_str, LC_C_LOCALE);
                                            ^
1 error generated.
make[2]: *** [CMakeFiles/uv.dir/src/unix/process.c.o] Error 1
```

This is just a name for NULL:
`#define LC_C_LOCALE				((locale_t)NULL)`

As specified in the man page:

> If a NULL locale_t is passed, the C locale will be used.

So we can use that value directly, rather than the convenience name add in 10.15
.
```suggestion
  version_major = atoi_l(version_major_str, NULL); /* Use LC_C_LOCALE */
```